src/component-library/Map/utils/helper.ts

```diff
@@ -0,0 +1,74 @@
+import maplibregl from "maplibre-gl";
+import { Geometry } from "geojson";
+import { ResultMarker } from "../ResultsMarkers";
+import { splitURIForNamespaceAndTerm } from "./strings";
+import geohash from "./geohash";
+
+export const getIconLabel = (uri: string) => {
+  const [namespace, term] = splitURIForNamespaceAndTerm(uri);
+  let label = namespace;
+  if (term !== "") {
+    label = term;
+  }
+  return label;
+};
+
+export const getLabelCharacters = (label: string): string =>
+  label.substring(0, 3);
+
+interface Feature {
+  type: "Feature";
+  geometry: Geometry;
+}
+
+export interface FeatureCollection {
+  type: "FeatureCollection";
+  features: Feature[];
+}
+
+export const checkIsValidPosition = (arr: any[]): boolean =>
+  Array.isArray(arr) &&
+  arr.length === 2 &&
+  !Array.isArray(arr[0]) &&
+  !Array.isArray(arr[1]);
+
+export const recursiveFlatten = (array: any[]) =>
+  checkIsValidPosition(array)
+    ? [array]
+    : array.reduce((acc, val) => acc.concat(recursiveFlatten(val)), []);
+
+export const getCoordinates = (geometry: Geometry) => {
+  if (geometry.type !== "Polygon" && geometry.type !== "MultiPolygon") {
+    throw new Error("Unsupported geometry type");
```

Comment:
Use `Error: Expected X, instead got Y` template
```suggestion
    throw new Error(`Unsupported geometry type: Expected "Polygon" or "MultiPolygon", instead got "${geometry.type}"`);
```